notebooks/matrix_tube_pipeline_seqcount_norm.ipynb

```diff
@@ -70,340 +75,461 @@
-    "#         'run_date': '2023-03-02', # date of MiSeq run\n",
-    "#         'Original Name': '' # leave empty\n",
     "    },\n",
-    "]\n"
-   ]
+    "]"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "cell_type": "code",
+   "source": [
+    "# CONSTANTS: Users, DO NOT CHANGE THESE\n",
+    "# values without consulting with tech team\n",
+    "SHEET_TYPE_VERSIONS = {\n",
+    "    _STANDARD_METAG_SHEET_TYPE: '102',  # version supporting SampleContext\n",
+    "    _ABSQUANT_SHEET_TYPE: '10'\n",
+    "}\n",
+    "\n",
+    "BIOINFO_BASE = {\n",
+    "    'ForwardAdapter': 'GATCGGAAGAGCACACGTCTGAACTCCAGTCAC',\n",
+    "    'ReverseAdapter': 'GATCGGAAGAGCGTCGTGTAGGGAAAGGAGTGT',\n",
+    "    'library_construction_protocol': 'Knight Lab Kapa HyperPlus',\n",
+    "    # The BarcodesAreRC value is no longer used, but is still checked for\n",
+    "    # by the validation while making the sample sheet, so put in a dummy value\n",
+    "    'BarcodesAreRC': 'True'\n",
+    "}\n",
+    "\n",
+    "WELL_COL = 'Well'\n",
+    "LIB_WELL_COL = 'Library Well'\n",
+    "\n",
+    "# Mask arrays for even and odd rows and columns\n",
+    "EVEN_ROWS = [x for x in range(16) if x % 2 == 0]\n",
+    "ODD_ROWS = [x for x in range(16) if x % 2 == 1]\n",
+    "EVEN_COLS = [x for x in range(24) if x % 2 == 0]\n",
+    "ODD_COLS = [x for x in range(24) if x % 2 == 1]"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "cell_type": "code",
+   "source": [
+    "def get_studies_attr_list(studies_dict, desired_key):\n",
+    "    return [x[desired_key] for x in studies_dict]"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": "### Part 1, Step 1: Read in sample accession files"
+  },
+  {
+   "metadata": {},
+   "cell_type": "code",
+   "source": [
+    "# read in the sample accession files\n",
+    "sample_accession_fps = get_studies_attr_list(\n",
+    "    studies_info, 'sample_accession_fp')\n",
+    "sample_accession_df = join_dfs_from_files(\n",
+    "    sample_accession_fps, [SAMPLE_NAME_KEY, TUBECODE_KEY])\n",
+    "sample_accession_df.shape"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "cell_type": "code",
+   "source": "sample_accession_df.head()",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": "### Part 1, Step 2: Read in the sample info from Qiita"
+  },
+  {
+   "metadata": {},
+   "cell_type": "code",
+   "source": [
+    "# read in the qiita metadata files\n",
+    "qiita_metadata_fps = get_studies_attr_list(studies_info, 'qiita_metadata_fp')\n",
+    "metadata_df = join_dfs_from_files(\n",
+    "    qiita_metadata_fps, [SAMPLE_NAME_KEY, QIITA_STUDY_ID_KEY], \n",
+    "    opt_cols_to_extract=['tube_id'], unique_cols=[SAMPLE_NAME_KEY])\n",
+    "metadata_df.shape"
```

Comment:
This will "print" the shape but not sure if we actually need that and/or if we should have an actual print or metadata.df.head() or something like that. 

Edit: BTW this applies to all cells where the last line is XX.shape.

Reply:
Yes, it is intended to print the shape.  Experience using the matrix tube notebook with the lab has shown that printing out the dataframe (or using head()) is insufficient to check that the expected dataframe was accurately created, because those display methods cut off rows and columns.  Shape makes it easy to see in an instant, e.g., "Oh, something must be wrong because I added two 10 line files but ended up with a dataframe with only 10 records" :)